shell/store/action-menu.js

```diff
@@ -111,6 +111,14 @@ export const mutations = {
     state.modalData = data;
   },
 
+  updateModalData(state, data) {
+    state.modalData = state.modalData || {};
+
+    data.forEach(({ key, value }) => {
+      state.modalData[key] = value;
```

Comment:
We want to modify properties instead of assign a full new object to modalData because it can trigger a full re-render of a modal which can reset component state unintentionally.